### DISCUSSION TITLE
Encoding for visualization in OpenLayers

### DISCUSSION THREAD

**Discussion by StefanBrand** [fiboa/specification#30] [Ideas]
Currently there is an [encoding for GeoParquet](https://github.com/fiboa/specification/blob/main/geoparquet/README.md). I see open issues for GeoPackage (#23) and FlatGeoBuf (#24). As far as I am aware neither of these are supported by OpenLayers. GeoJSON is not suitable to stream portions of the dataset for viewing.

What do you think about an encoding for a viewing-optimized format, e.g. Vector Tiles?

**Comment by m-mohr** [comment]
We currently convert all datasets that we publish also as PMTiles: https://beta.source.coop/fiboa/
It's a simple conversion from GeoParquet via GeoJSON to PMTiles through tippecanoe. 

I think we could make this an official encoding and write the additional fiboa metadata to https://github.com/protomaps/PMTiles/blob/main/spec/v3/spec.md#5-json-metadata
On the other hand, PMTiles should probably not be the main encoding in which data is delivered, maybe it just an extension to visualize the data?
Unfortunately, the PMTiles creation also heavily depends on tippecanoe and there's no easy way to e.g. export to PMTiles from GeoPandas.

**Reply by StefanBrand** [reply]
I was more thinking in the direction of on-the-fly generation of tiles from a PostGIS database. We are using [`ST_AsMVT`](https://postgis.net/docs/manual-dev/en/ST_AsMVT.html) to encode fields as Mapbox Vector Tiles.

Then again, if I understand correctly, OGC API Tiles allows for a [GeoJSON encoding](https://docs.ogc.org/is/20-057/20-057.html#_geojson_encoding) and then fiboa might already be compatible with this use-case.

**Reply by m-mohr** [reply]
Yeah, indeed. We primarily only define how the properties are named and the semantics of the values so you can easily make fiboa data available via OGC API - Tiles and ST_AsMVT as long as the underlying Features were fiboa compliant during ingestion, I think.

**Comment by StefanBrand** [comment]
The `flatgeobuf` package actually supports OpenLayers and there is are examples of a FlatGeoBuf file visualized in an `ol` map: https://flatgeobuf.org/examples/openlayers/
